### DISCUSSION TITLE
Querying item inside list

### DISCUSSION THREAD

**Discussion by markosalazar** [infinispan/infinispan#12701] [Q&A] [answered]
Hello everyone, 

Using infinispan 15.0.3 with protostream 5.3. Have this simple example I am trying to implement a query on.

Proto file:

    // File name: test.proto
    // Generated from : test.proto
    syntax = "proto2";
    package test;

    /**
     * @Indexed
     */
    message Book {
           /**
            * @Basic
            */
           optional string title = 1;
           
           optional string description = 2;
           
           optional int32 publicationYear = 3 [default = 0];
    }

    /**
     * @Indexed
     */
    message BookList {
       
           /**
            * @List({@org.infinispan.api.annotations.indexing.Embedded})
            * @List({@org.infinispan.api.annotations.indexing.Basic})
            */
           repeated Book bookList = 1;
    }

I wanted to return a list of books that is stored within BookList (embedded query) or even the BookList object itself. But I cannot build a cache with BookList as an indexed entity, because it doesn't like the List field as an indexed field, throws this error:

The configured indexed-entity type 'lmo.BookList' must be indexed. Please annotate it with @Indexed and make sure at least one field has the @Field annotation

Also tried annotating the indexed fields like this but still seeing the issue:
@ProtoComment("@Indexed")
@ProtoComment("@Field(index = Index.YES,store = Store.YES, analyze = Analyze.Yes)")

Any feedback on this ? Would greatly appreciate it, thanks!

**Comment by markosalazar** [comment]
Hello everyone, any idea on this ? Or is this not supported currently?

**Comment by faxami** [comment, marked as answer]
Hi @markosalazar. Why are you using @List annotation? I would annotate the  `bookList` field simply with `@Embedded` annotation. Please let us know if it worked. Thanks
